fiona/ogrext.pyx

```diff
@@ -1287,6 +1295,37 @@ def _remove(path, driver=None):
         raise RuntimeError("Failed to remove data source {}".format(path))
```

Comment:
Instead of `RuntimeError` let's raise a very specific error about data source deletion. We'll need a new exception class derived from `fiona.error.FionaError`. I'm not sure what the best name will be.